src/Microsoft.DocAsCode.Build.TableOfContents/TocDocumentProcessor.cs

```diff
@@ -35,42 +28,14 @@ public override ProcessingPriority GetProcessingPriority(FileAndType file)
             return ProcessingPriority.NotSupported;
         }
 
-        public override FileModel Load(FileAndType file, ImmutableDictionary<string, object> metadata)
-        {
-            var filePath = file.FullPath;
-            var toc = Utility.LoadSingleToc(filePath);
-
-            var repoDetail = GitUtility.TryGetFileDetail(filePath);
-            var displayLocalPath = PathUtility.MakeRelativePath(EnvironmentContext.BaseDirectory, file.FullPath);
-
-            // todo : metadata.
-            return new FileModel(file, toc)
-            {
-                Uids = new[] { new UidDefinition(file.File, displayLocalPath) }.ToImmutableArray(),
-                LocalPathFromRoot = displayLocalPath
-            };
-        }
-
-        public override SaveResult Save(FileModel model)
+        public override void RegisterTocToContext(TocItemViewModel toc, FileModel model, IDocumentBuildContext context)
```

Comment:
reload TocItemViewModel will break original bahavior, add back, later only child item will call register method so it's ok.

---
In reply to: [121039829](https://github.com/dotnet/docfx/pull/1746#discussion_r121039829) [](ancestors = 121039829)